### PR TITLE
:hammer: Get hash for API_Objects:serverCreateObject and Fix vehicle won't spawn and Add _NCS.Ped:serverSpawn

### PR DESCRIPTION
**Pull request by ZAKIO59**
In order to be coordinated with other functions.

### CODE REVIEW

**Review by MoskalykA — APPROVED** [review]

**Comment by Ekinoxx0** [comment]
It seems like these function would preferebly accept both depending on the type

```lua
-- Accept a var string or number
assert(type(var) == 'string' or type(var) == 'number')

if type(var) == 'string' then -- Only do this if needs a convertion
    var = GetHashKey(var)
end

CreateObject(var, ....)
```

This will allow pre-runtime compilation of hashes like this one : \`hashed_string\`

This comment could also apply to the vehicle part of this fix, accepting vector3/vector4/array ?

**Review by MoskalykA — CHANGES_REQUESTED** [review]
Use " please

**Comment by ZAKIO59** [comment]
It's good ?

**Comment by MoskalykA** [comment]
> It's good ?

This should be done in the other files.

**Comment by ZAKIO59** [comment]
> > It's good ?
> 
> This should be done in the other files.

My bad, I think it's good.

File: src/api/objects/server/serverCreateObject.lua [hunk 1]
```diff
@@ -7,6 +7,12 @@
 ---@return void
 ---@public
 function API_Objects:serverCreateObject(props, x, y, z, h)
+    assert(type(props) == "string" or type(props) == "number")
+
+    if type(props) == "string" then
+        props = GetHashKey(props)
+    end
+
     local object = CreateObject(props, x, y, z - 1, true)
```
Comment (MoskalykA) [inline src/api/objects/server/serverCreateObject.lua:16]: Why z - 1 ?

File: src/api/ped/server/spawnPed.lua [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+---@public
+function API_Ped:serverSpawn(modelName, x, y, z, heading, cb)
+    assert(type(modelName) == "string" or type(modelName) == "number")
+
+    if type(modelName) == "string" then
+        modelName = GetHashKey(modelName)
+    end
+
+    local vehicle = CreatePed(1, modelName, x, y, z, heading)
+
```
Comment (MoskalykA) [inline src/api/ped/server/spawnPed.lua:18]: Unnecessary space

File: src/api/ped/server/spawnPed.lua [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+---@param x number
+---@param y number
+---@param z number
+---@param heading number
+---@param cb function
+---@return void
+---@public
+function API_Ped:serverSpawn(modelName, x, y, z, heading, cb)
+    assert(type(modelName) == "string" or type(modelName) == "number")
+
```
Comment (MoskalykA) [inline src/api/ped/server/spawnPed.lua:12]: Unnecessary space

File: src/api/objects/server/serverCreateObject.lua [hunk 4]
```diff
@@ -7,6 +7,12 @@
 ---@return void
 ---@public
 function API_Objects:serverCreateObject(props, x, y, z, h)
+    assert(type(props) == "string" or type(props) == "number")
+
```
Comment (MoskalykA) [inline src/api/objects/server/serverCreateObject.lua:11]: Unnecessary space

File: src/api/vehicles/client/spawnVehicle.lua [hunk 5]
```diff
@@ -7,6 +7,12 @@
 ---@public
 function API_Vehicles:spawn(modelName, coords, heading, cb, clientSide)
     CreateThread(function()
+        assert(type(modelName) == "string" or type(modelName) == "number")
+
```
Comment (MoskalykA) [inline src/api/vehicles/client/spawnVehicle.lua:11]: Unnecessary space

File: src/api/vehicles/server/spawnVehicle.lua [hunk 6]
```diff
@@ -1,11 +1,20 @@
...
+---@param z number
+---@param heading number
 ---@param cb function
 ---@return void
 ---@public
-function API_Vehicles:serverSpawn(modelName, coords, cb)
-    local vehicle = CreateVehicle(GetHashKey(modelName), coords)
+function API_Vehicles:serverSpawn(modelName, x, y, z, heading, cb)
+    assert(type(modelName) == "string" or type(modelName) == "number")
+
```
Comment (MoskalykA) [inline src/api/vehicles/server/spawnVehicle.lua:12]: Unnecessary space

**Review by MoskalykA — CHANGES_REQUESTED** [review]

File: src/api/objects/server/serverCreateObject.lua (hunk 1 above)
Comment (ZAKIO59) [reply src/api/objects/server/serverCreateObject.lua:16]: Idk, it wasn't done by me. I'll delete it

File: src/api/objects/server/serverCreateObject.lua [hunk 7]
```diff
@@ -7,7 +7,12 @@
 ---@return void
 ---@public
 function API_Objects:serverCreateObject(props, x, y, z, h)
```
Comment (MoskalykA) [inline src/api/objects/server/serverCreateObject.lua:9]:
```suggestion
function API_Objects:createObject(props, x, y, z, heading)
```

File: src/api/ped/server/spawnPed.lua [hunk 8]
```diff
@@ -0,0 +1,22 @@
+---serverSpawn
+---@param modelName string
+---@param x number
+---@param y number
+---@param z number
+---@param heading number
+---@param cb function
```
Comment (MoskalykA) [inline src/api/ped/server/spawnPed.lua:7]:
```suggestion
---@param cb? function
```

File: src/api/ped/server/spawnPed.lua [hunk 9]
```diff
@@ -0,0 +1,22 @@
+---serverSpawn
```
Comment (MoskalykA) [inline src/api/ped/server/spawnPed.lua:1]:
```suggestion
---spawn
```

File: src/api/ped/server/spawnPed.lua [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+---@param modelName string
+---@param x number
+---@param y number
+---@param z number
+---@param heading number
+---@param cb function
+---@return void
+---@public
+function API_Ped:serverSpawn(modelName, x, y, z, heading, cb)
```
Comment (MoskalykA) [inline src/api/ped/server/spawnPed.lua:10]:
```suggestion
function API_Ped:spawn(modelName, x, y, z, heading, cb)
```

File: src/api/vehicles/server/spawnVehicle.lua [hunk 11]
```diff
@@ -1,11 +1,19 @@
 ---serverSpawn
```
Comment (MoskalykA) [inline src/api/vehicles/server/spawnVehicle.lua:1]:
```suggestion
---spawn
```

File: src/api/vehicles/server/spawnVehicle.lua [hunk 12]
```diff
@@ -1,11 +1,19 @@
...
 ---@param modelName string
----@param coords table (vector4)
+---@param x number
+---@param y number
+---@param z number
+---@param heading number
 ---@param cb function
```
Comment (MoskalykA) [inline src/api/vehicles/server/spawnVehicle.lua:7]:
```suggestion
---@param callback function
```

File: src/api/vehicles/server/spawnVehicle.lua [hunk 13]
```diff
@@ -1,11 +1,19 @@
...
 ---@return void
 ---@public
-function API_Vehicles:serverSpawn(modelName, coords, cb)
-    local vehicle = CreateVehicle(GetHashKey(modelName), coords)
+function API_Vehicles:serverSpawn(modelName, x, y, z, heading, cb)
```
Comment (MoskalykA) [inline src/api/vehicles/server/spawnVehicle.lua:10]:
```suggestion
function API_Vehicles:spawn(modelName, x, y, z, heading, callback)
```

File: src/api/vehicles/server/spawnVehicle.lua [hunk 14]
```diff
@@ -1,11 +1,19 @@
...
+    assert(type(modelName) == "string" or type(modelName) == "number")
+    if type(modelName) == "string" then
+        modelName = GetHashKey(modelName)
+    end
+
+    local vehicle = CreateVehicle(modelName,  x, y, z, heading)
```
Comment (MoskalykA) [inline src/api/vehicles/server/spawnVehicle.lua:16]:
```suggestion
    local vehicle = CreateVehicle(modelName, x, y, z, heading)
```

File: src/api/vehicles/client/spawnVehicle.lua [hunk 15]
```diff
@@ -7,6 +7,11 @@
 ---@public
 function API_Vehicles:spawn(modelName, coords, heading, cb, clientSide)
```
Comment (MoskalykA) [inline src/api/vehicles/client/spawnVehicle.lua:8]:
```suggestion
function API_Vehicles:spawn(modelName, coords, heading, callback, clientSide)
```

File: src/api/vehicles/client/spawnVehicle.lua [hunk 16]
```diff
@@ -7,6 +7,11 @@
...
     CreateThread(function()
```
Comment (MoskalykA) [inline src/api/vehicles/client/spawnVehicle.lua:9]: Remove CreateThread

**Review by MoskalykA — CHANGES_REQUESTED** [review]

**Comment by ZAKIO59** [comment]
It's good ?

**Comment by MoskalykA** [comment]
There are a lot of things I didn't mention that you need to change.

**Comment by ZAKIO59** [comment]
It's good ?

**Comment by MoskalykA** [comment]
> It's good ?

Make git pull please

**Comment by ZAKIO59** [comment]
> > It's good ?
> 
> Make git pull please
>
> I can't

**Pull request closed without merging**